crates/transaction-pool/src/pool/best.rs

```diff
@@ -131,7 +131,12 @@ impl<T: TransactionOrdering> BestTransactions<T> {
     /// Checks for new transactions that have come into the `PendingPool` after this iterator was
     /// created and inserts them
     fn add_new_transactions(&mut self) {
+        let mut new_txs = Vec::new();
         while let Some(pending_tx) = self.try_recv() {
+            new_txs.push(pending_tx);
+        }
+        new_txs.sort_by_key(|tx| tx.transaction.nonce());
```

Comment:
this looks a bit weird, why can we no longer process txs one by one?

how many txs are buffered in `new_txs` is arbitrary, so doesn't make sense that we have to sort this arbitrary chunk

Reply:
updated `BestTransactions`, it now holds the following:
https://github.com/paradigmxyz/reth/blob/311a10d7b950c11aa3a970db11fdb4c0ac06d63b/crates/transaction-pool/src/pool/best.rs#L77-L83

when inserting, we make sure to replace independent tx if the nonce of tx being added is lower than the lowest we know for the given sender

that way we still don't make assumptions on order of transactions and don't have to sort everything